src/sql/platform/connection/common/rpcConnectionProfile.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { deepClone } from 'vs/base/common/objects';
+
+
+// Concrete implementation of the connection.ConnectionProfile interface
+
+/**
+ * A concrete implementation of a RPC friendly ConnectionProfile
+ */
+export class RpcConnectionProfile implements azdata.connection.ConnectionProfile {
```

Comment:
1. I am not sure what does RPC friendly means, you can only pass data between RPC endpoints, not the class itself.

2. Isn't there already an interface or class that is the same as azdata.connection.ConnectionProfile in core?
3. duplicate comments above.

Reply:
1. Maybe I should say, used for passing data of a connection profile between RPC points.
2. There is a MockConnectionProfile in the datavirtualization extension, but that isn't usable for us, so I had to create a new class implementing the ConnectionProfile interface.

Reply:
You can't pass classes over RPC though, that's often where you end up getting those circular reference errors. (note that sometimes it'll seem to work fine, but that's just a quirk of how classes in JS work and is likely to break at some later point so should be avoided)

That being said, we already pass over the ConnectionProfile class in a bunch of other places so it's fine just to do something like this : https://github.com/Microsoft/azuredatastudio/blob/8a3d08f0dec1cdf47d7fc442f90fb2227b828375/src/sql/workbench/api/browser/mainThreadConnectionManagement.ts#L104 where you just create a simple object and assign properties to it. 

(this is still not really what we should be doing overall, since it loses all the class functionality on the other side. But fixing that is way beyond the scope of this PR and there isn't a much better solution for the time being so it's fine to just do something similar)

Reply:
Yep it works, gonna clean it.